### PR TITLE
Adding Support for Service Configuration for appliances

### PR DESCRIPTION
**Pull request by Bhavneet-Sharma**
# Description
Adding Support for Service Configuration for appliances using service_config module

# GitHub Issues
List the GitHub issues impacted by this PR:

| GitHub Issue # |
| -------------- |
| |

# Checklist:

- [x] I have performed a self-review of my own code to ensure there are no formatting, pep8, linting, or security issues
- [x] I have performed Ansible Sanity test using --docker default
- [x] I have verified that new and existing unit tests pass locally with my changes
- [x] I have not allowed coverage numbers to degenerate
- [x] I have maintained at least 90% code coverage
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] Backward compatibility is not broken

# How Has This Been Tested?
Please describe the tests that you ran to verify your changes. Please also list any relevant details for your test configuration

- [x] Functional Test
- [x] Unit Test

### CODE REVIEW

**Comment by ansible-collections-svc** [comment]
Can one of the admins verify this patch?

**Review by sachin-apa — APPROVED** [review]

**Review by felixs88 — DISMISSED** [review]

**Review by meenakshidembi691 — DISMISSED** [review]

File: docs/modules/service_config.rst [hunk 1]
```diff
@@ -0,0 +1,168 @@
...
+
+
+
+Requirements
+------------
+The below requirements are needed on the host that executes this module.
+
+- A Dell PowerStore storage system version 3.0.0.0 or later.
+- Ansible-core 2.13 or later.
+- PyPowerStore 2.0.0.
```
Comment (trisha-dell) [inline docs/modules/service_config.rst:25]: change this to 3.0.0

File: plugins/module_utils/storage/dell/libraries/configuration.py [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+            appliance_name (str): The name of the appliance.
+        Returns:
+            dict: The details of the appliance.
+        Raises:
+            Exception: If an error occurs while getting the appliance details.
+        """
+
+        try:
+            # Log the appliance details being retrieved
+            log_msg = f"Getting Appliance details id: {appliance_id} or " \
```
Comment (trisha-dell) [inline plugins/module_utils/storage/dell/libraries/configuration.py:45]: details **with** id

File: plugins/modules/service_config.py [hunk 3]
```diff
@@ -0,0 +1,317 @@
...
+      appliance_id:
+        description:
+        - Specifies the appliance id.
+        - Mutually exclusive with appliance_name.
+        type: str
+      is_ssh_enabled:
+        description:
+        - Whether ssh will be enabled/disabled on specified appliance.
+        type: bool
+        required: true
```
Comment (trisha-dell) [inline plugins/modules/service_config.py:50]: this param will not be needed for get operation, hence setting it as required is not ideal.

File: plugins/modules/service_config.py [hunk 4]
```diff
@@ -0,0 +1,317 @@
...
+  retrieving, and updating Service config.
+
+extends_documentation_fragment:
+  - dellemc.powerstore.powerstore
+
+author:
+- Bhavneet Sharma(@Bhavneet-Sharma) <ansible.team@dell.com>
+
+options:
+  service_config:
```
Comment (trisha-dell) [inline plugins/modules/service_config.py:30]: change this to ssh_access or update the example. whichever is correct.

File: plugins/modules/service_config.py (hunk 3 above)
Comment (Bhavneet-Sharma) [reply plugins/modules/service_config.py:50]: This param is required only when service_config list of dict param passed. So it required for service_config param, and not required in Get operation.

**Review by felixs88 — APPROVED** [review]

**Review by trisha-dell — APPROVED** [review]

**Pull request merged**